llvm/lib/CAS/OnDiskHashMappedTrie.cpp

```diff
@@ -18,8 +18,6 @@
 #include "llvm/Support/Path.h"
 #include "llvm/Support/raw_ostream.h"
 
-#if LLVM_ENABLE_ONDISK_CAS
-
 using namespace llvm;
 using namespace llvm::cas;
 
```

Comment:
The main goal of the macro/cmake configuration is to guard the static_assert below, otherwise it is going to fail to build on 32 bit arch. If the goal is thoroughly get the code to build, everything probably compiles fine on all platforms with those assertion removed.

Reply:
Github won't let me comment on the line that is not in the diff. It is basically the line after here but it is not shown in snippet.

Reply:
> The main goal of the macro/cmake configuration is to guard the static_assert below, otherwise it is going to fail to build on 32 bit arch.

That's not what controls the configuration though, in CMake it is
```
if(UNIX)
  set(LLVM_ENABLE_ONDISK_CAS_default ON)
else()
  set(LLVM_ENABLE_ONDISK_CAS_default OFF)
endif()
```
which indicates it's only disabled by default on windows, the architecture is not checked.

Reply:
Whether it is enabled or not by default is not the point, the point is that it is a switch you can turn off if it doesn't work on your platform.

So technically, this PR adds on swift branch from external developer that wasn't landed on other branches: https://github.com/apple/llvm-project/pull/6098

Reply:
Thanks for pointing out that commit! Generally it would be better that we direct people to make changes on `experimental/cas/main` and cherry-pick on `next` (or vice-versa), we should not be diverging.

Reply:
I cherry-picked https://github.com/apple/llvm-project/pull/6098 to this PR to get this branch in-sync with `next` and changed `OnDiskHashMappedTrie.cpp` to guard its implementation with `LLVM_ENABLE_ONDISK_CAS` while still keeping the macro out of the header.